library/src/main/java/ch/decent/sdk/model/Message.kt

```diff
@@ -0,0 +1,56 @@
+package ch.decent.sdk.model
+
+import ch.decent.sdk.crypto.Address
+import ch.decent.sdk.crypto.Credentials
+import ch.decent.sdk.crypto.ECKeyPair
+import ch.decent.sdk.utils.decryptAesWithChecksum
+import ch.decent.sdk.utils.secret
+import ch.decent.sdk.utils.unhex
+import com.google.gson.annotations.SerializedName
+import org.threeten.bp.LocalDateTime
+import java.math.BigInteger
+
+data class Message(
+    @SerializedName("id") val id: ChainObject,
+    @SerializedName("created") val created: LocalDateTime,
+    @SerializedName("sender") val sender: ChainObject,
+    @SerializedName("sender_pubkey") val senderAddress: Address,
+    @SerializedName("receivers_data") val receiversData: List<MessageReceiver>,
+    @SerializedName("text") val text: String
+) {
+
+  private fun decryptOrEmpty(keyPair: ECKeyPair, payload: MessageReceiver, address: Address = payload.receiverAddress) = try {
+    decryptAesWithChecksum(keyPair.secret(address, payload.nonce), payload.data.unhex())
+  } catch (ex: Exception) {
+    ""
+  }
+
+  fun decrypt(credentials: Credentials): List<Pair<ChainObject, String>> =
+      when {
+        credentials.account == sender -> receiversData.map { it.receiver to decryptOrEmpty(credentials.keyPair, it) }
+        credentials.account != sender -> receiversData
+            .filter { it.receiver == credentials.account }
```

Comment:
is this filter here because writing messages to yourself is forbidden by DCore? I can imagine there are valid use cases where you want to send messages to yourself.

Reply:
I think sending message to yourself should be supported.

Reply:
i think it is forbidden, but that is not the point
you can send message to multiple receivers with one `SendMessageOperation`
if you are the receiver you need to filter out other receivers, since everybody gets the same `Message` response